ci/cloudfront-orchestration/waf/template.yaml

```diff
@@ -0,0 +1,184 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: >
+  Stack to create a WAF to attach to the Cloudfront distribution for the OIDC API
+
+Parameters:
+  Environment:
+    Type: String
+    AllowedValues: ["dev", "build", "staging", "integration", "production"]
+  SmokeTestClientId:
+    Type: String
+    Default: none
+
+Conditions:
+  IsStaging:
+    !Equals [ staging, !Ref Environment ]
+  IsProduction:
+    !Equals [ production, !Ref Environment ]
+
+Resources:
+  WAF:
+    Type: AWS::WAFv2::WebACL
+    Properties:
+      Name: !Sub "${Environment}-oidc-cloudfront-waf"
+      Scope: CLOUDFRONT
+      DefaultAction:
+        Allow: {}
+      Rules:
+        - Name: !Sub "${Environment}-oidc-waf-rate-based-rule"
+          Action:
+            Block: {}
+          Priority: 1
+          Statement:
+            RateBasedStatement:
+              Limit: !If [IsStaging, 600000, 3600]
+              AggregateKeyType: "IP"
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: !Sub "${Environment}OidcWafMaxRequestRate"
+            SampledRequestsEnabled: true
+        - Name: !Sub "${Environment}-oic-common-rule-set"
+          OverrideAction:
+            None: {}
+          Priority: 2
+          Statement:
+            ManagedRuleGroupStatement:
+              Name: "AWSManagedRulesCommonRuleSet"
+              VendorName: "AWS"
+              RuleActionOverrides:
+                - Name: "GenericRFI_QUERYARGUMENTS"
+                  ActionToUse:
+                    Count: {}
+                - Name: "GenericRFI_BODY"
+                  ActionToUse:
+                    Count: {}
+                - Name: "NoUserAgent_HEADER"
+                  ActionToUse:
+                    Count: {}
+                - Name: "EC2MetaDataSSRF_QUERYARGUMENTS"
+                  ActionToUse:
+                    Count: {}
+                - Name: "SizeRestrictions_QUERYSTRING"
+                  ActionToUse:
+                    Count: {}
+                - !If
+                  - IsProduction
+                  - Name: "EC2MetaDataSSRF_BODY"
+                    ActionToUse:
+                      Count: {}
+                  - !Ref AWS::NoValue
```

Comment:
Is this not the opposite logic to the terraform? Also would be good to do testing around this as we've had issues around it in the past

Reply:
I need to take a look at this. The logic in terraform makes no sense because `(NOT prod) OR (NOT sandpit)` is always true...

Reply:
That's a very good point and maybe why we've had issues in the past changing it...